test/jdk/java/awt/Desktop/ActiveWindowCloseTest.java

```diff
@@ -0,0 +1,124 @@
+import java.awt.event.KeyEvent;
+import java.lang.reflect.InvocationTargetException;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8297095
+ * @summary To determine the quit request of the application
+ * @run main ActiveWindowCloseTest
+ */
+public class ActiveWindowCloseTest extends JFrame implements QuitHandler {
+    private static ActiveWindowCloseTest frame = null;
+    private static volatile int count = 0;
+    private static Robot robot;
+    private static boolean passed = false;
+
+    private ActiveWindowCloseTest(String name) {
+        super(name);
+    }
+
+    @Override
+    public void handleQuitRequestWith(QuitEvent e, QuitResponse response) {
+        if (count++ == 2) {
+            System.out.println("Received performQuit()");
+            response.performQuit();
+        } else {
+            System.out.println("Received cancelQuit()");
+            response.cancelQuit();
+        }
+    }
+
+    private static void createAndShow() {
+        try {
+            frame = new ActiveWindowCloseTest("Close windows");
+            Desktop.getDesktop().setQuitHandler(frame);
+            Desktop.getDesktop().disableSuddenTermination();
+            Desktop.getDesktop()
+                .setQuitStrategy(QuitStrategy.CLOSE_ALL_WINDOWS);
+            frame.pack();
+            frame.setLocationRelativeTo(null);
+            frame.setVisible(true);
+            passed = true;
+        } catch (UnsupportedOperationException usoe) {
+            System.out.println("Test skipped, as the action "
+                + "is not supported on the current platform!");
+        }
+    }
+
+    public static void main(String args[])
+        throws AWTException, InvocationTargetException, InterruptedException {
+        boolean isMac = false;
+        try {
+            if (System.getProperty("os.name").toLowerCase().contains("os x")) {
+                isMac = true;
+            }
+            SwingUtilities.invokeAndWait(ActiveWindowCloseTest::createAndShow);
+
+            if (passed) {
+                robot = new Robot();
+                robot.setAutoDelay(50);
+                robot.setAutoWaitForIdle(true);
+                robot.waitForIdle();
+
+                for (int i = 0; i < 3; i++) {
+                    if (frame.isActive()) {
+                        if (isMac) {
+                            robot.keyPress(KeyEvent.VK_META);
+                            robot.keyPress(KeyEvent.VK_Q);
+                            robot.keyRelease(KeyEvent.VK_Q);
+                            robot.keyRelease(KeyEvent.VK_META);
+                        } else {
+                            robot.keyPress(KeyEvent.VK_ALT);
+                            robot.keyPress(KeyEvent.VK_F4);
+                            robot.keyRelease(KeyEvent.VK_F4);
+                            robot.keyRelease(KeyEvent.VK_ALT);
+                        }
+                        robot.waitForIdle();
```

Comment:
Adding an additional delay (`robot.delay(200)`) will make sure that all the preceding events are processed.